### ISSUE TITLE
allow click and drag to select multiple connections and revise "shift"

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#2221]
So far you cannot select multiple connections. 

Click on an empty spot and drag only selects boxes (and their connections). There is no way to only select cords. So one request would be to be able to click and drag to select just cords/connections.

For boxes, we're allowed to select multiple and individual items with "shift+click", but this is not possible with cords, as it just swaps connection. I think it makes sense to be able to select more than one cord with shift+click and we can use a different key, like 'alt+click' to 'alternate' and swap connections (**EDIT: Shift+double click makes better sense I think, and it's how PlugData does it now**).

Being able to select multiple connections is nice to delete more than one at a time.

**Comment by umlaeute** [comment]
The current data structures only have a single int (the index of the connection).
Since the structure is semi-private, changing this might break binary compat.

Apart from that, I don't see much point on being able to select more than one connections at a time: the argument that one can then delete multiple connections seems especially weak. (The annoying part is that you have to select the connections, not that you have to press <kbd>Delete</kbd>).
The only nice thing about deleting them at once is being able to *undo* all of them with a single keystroke, but I'm not convinced that this warrants breaking binary compat.

**Comment by porres** [comment]
think of unpack and pack with a zillion connections and you want to disconnect them all. Now you can select one or another and break all connections for either, but say you chose unpack and don't want to loose its input connection. There's no way to do this currently and I've seen myself desiring for such a thing.

**Comment by umlaeute** [comment]
yes, but your solution is basically doing <kbd>Shift</kbd>+<kbd>Click</kbd> a zillion times and then do a <kbd>Delete</kbd>.
i don't see this as significantly better than doing <kbd>Click</kbd>-<kbd>Delete</kbd> a zillion times.

so what is the point?

**Comment by porres** [comment]
> yes, but your solution is basically doing Shift+Click a zillion times

not really, I also suggested that we could be able to click and drag to make a rectangular selection that only includes patch cords. There is no way to do that yet. 

Shift+click would be also useful, but for two or three (not a zillion) connections that cannot be singled out by a rectangular selection

**Comment by Spacechild1** [comment]
The description is a bit confusing. I think there are actually *two* requests:

1. allow to click and drag to select just cords/connections.
2. select several cords with shift + Click

I think 1. would definitely make sense from a UX point of view (though I have no idea how hard it would be to implement).

I agree that Shift+click is not really *necessary*, but if we implement click+drag it would be more consistent with other selections. *If* I can drag my mouse to select something, I really expect Shift+Click to (de)select individual elements.

**Comment by umlaeute** [comment]
indeed i missed the drag-selecting of connections.

**Comment by porres** [comment]
Hi, I'd like to revise my request.

PlugData requires shift and double click for swapping connections, I think that makes good sense.

And then shift+click would allow us to select specific connections to delete them

moreover, the click and drag request still remains to select just connections
